pkg/provider/credvalidate/validator_stackit.go

```diff
@@ -0,0 +1,54 @@
+/*
+SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package credvalidate
+
+import (
+	"context"
+	"maps"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/gardener/gardenctl-v2/pkg/provider/common/allowpattern"
+	"github.com/gardener/gardenctl-v2/pkg/provider/common/credvalidate"
+)
+
+// STACKITValidator implements the common validator interface for STACKIT.
+type STACKITValidator struct {
+	openstack *OpenStackValidator
+	stackit   *credvalidate.BaseValidator
+}
+
+var _ credvalidate.Validator = &STACKITValidator{}
+
+// NewSTACKITValidator creates a new STACKIT validator.
+func NewSTACKITValidator(ctx context.Context) *STACKITValidator {
+	return &STACKITValidator{
+		openstack: NewOpenStackValidator(ctx, []allowpattern.Pattern{}),
+		stackit:   credvalidate.NewBaseValidator(ctx, []allowpattern.Pattern{}),
+	}
+}
+
+// ValidateSecret validates OpenStack credentials from a Kubernetes secret.
+func (v *STACKITValidator) ValidateSecret(secret *corev1.Secret) (map[string]interface{}, error) {
+	fields := credvalidate.FlatCoerceBytesToStringsMap(secret.Data)
+	registry := map[string]credvalidate.FieldRule{
+		"serviceaccount.json": {Required: true, Validator: nil, NonSensitive: false},
+		"project-id":          {Required: true, Validator: nil, NonSensitive: false},
```

Comment:
make sure to always have a validator implementation set. If the schema is not well defined / rather open on what values are allowed, at least check 
- that only printable characters were used like [here](https://github.com/gardener/gardenctl-v2/blob/6899190191c6c8053961bef423a8598de7df7f2e/pkg/provider/credvalidate/validator_openstack.go#L248)
- for sane max length like [here](https://github.com/gardener/gardenctl-v2/blob/6899190191c6c8053961bef423a8598de7df7f2e/pkg/provider/credvalidate/validator_openstack.go#L253)

Reply:
I added some basic validation.

I am wondering about the validation here. If a secret is not valid a shoot is already created with that invalid credentials. It makes sense to validate this in the provider-extensions but what is the purpose of validate it here in the gardenctl? Do I miss something?